src/AccountServices.js

```diff
@@ -0,0 +1,147 @@
+    this._storage.remove_client_data(event.sender);
+    this._storage.remove_timeline_room(event.sender);
+    this._twitter.detach_user_stream(event.sender);
+    intent.sendMessage(event.room_id,{"body":"Your account (if it was linked) is now unlinked from Matrix.","msgtype":"m.text"});
+  }
+  else if(isNumber(event.content.body)){
+    this._storage.get_client_data(event.sender).then((client_data) => {
+      var pin = event.content.body;
+      log.info("Handler.AccountServices","User sent a pin in to auth with.");
+      if(client_data == null){
+        intent.sendMessage(event.room_id,{"body":"You must request access with 'link account' first.","msgtype":"m.text"});
+        return;
+      }
+      this._oauth_getAccessToken(pin,client_data,event.sender).then((profile) => {
+        intent.sendMessage(event.room_id,{"body":"All good. You should now be able to use your Twitter account on Matrix.","msgtype":"m.text"});
+        this._twitter.create_user_timeline(event.sender,profile);
+        this._twitter.attach_user_stream(event.sender);
+      }).catch(err => {
+        intent.sendMessage(event.room_id,{"body":"We couldn't verify this PIN :(. Maybe you typed it wrong or you might need to request it again.","msgtype":"m.text"});
+        log.error("Handler.AccountServices","OAuth Access Token Failed:%s", err);
+      });
+    });
+  }
+}
+
+AccountServices.prototype._oauth_getAccessToken = function (pin,client_data,id) {
+  return new Promise((resolve,reject) => {
+    if(client_data && client_data.oauth_token != "" && client_data.oauth_secret != ""){
+      this._oauth.getOAuthAccessToken(client_data.oauth_token, client_data.oauth_secret, pin, (error,access_token,access_token_secret) =>{
+        if(error){
+          reject(error.statusCode + ": " + error.data);
+        }
+        client_data.access_token = access_token;
+        client_data.access_token_secret = access_token_secret;
+        client = new Twitter({
+          consumer_key: this._app_auth.consumer_key,
+          consumer_secret: this._app_auth.consumer_secret,
+          access_token_key: client_data.access_token,
+          access_token_secret: client_data.access_token_secret
+        });
+        client.get("account/verify_credentials",(error,profile) =>{
+          if(error){
+            log.error("Handler.AccountServices","We couldn't authenticate with the supplied access token for "+ id +". Look into this.");
+            reject("Twitter account could not be authenticated.");
+            return;
+          }
+          this._storage.set_client_data(id,profile.id,client_data);
+          resolve(profile);
+        });
+      });
+    }
+    else {
+      reject("User has no associated token request data");
+    }
+  });
+}
+
+AccountServices.prototype._oauth_getUrl = function(event,id){
+  return new Promise((resolve,reject) => {
+    this._oauth.getOAuthRequestToken({"x_auth_access_type":"dm"},(error, oAuthToken, oAuthTokenSecret, results) => {
```

Comment:
I don't know what `{"x_auth_access_type":"dm"}` is specifying - and most likely anyone reading this won't either. A comment to the API reference would make things nicer for readers.